riscv/src/compiler.rs

```diff
@@ -1393,13 +1393,13 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
         }
         "ecall" => {
             assert!(args.is_empty());
-            vec!["x10 <=X= ${ (\"input\", std::prover::eval(x10)) };".to_string()]
+            vec!["x10 <=X= ${ std::prover::Query::Input(std::prover::eval(x10))) };".to_string()]
```

Comment:
```suggestion
            vec!["x10 <=X= ${ std::prover::Query::Input(std::prover::eval(x10)) };".to_string()]
```

Reply:
Error during type inference:
Error checking sub-expression std::prover::eval(main.x10):
Expected type: int
Inferred type: fe
Cannot unify types fe and int


looks like this still needs type conversion